openlibrary/plugins/openlibrary/js/service-worker.js

```diff
@@ -0,0 +1,104 @@
+import { ExpirationPlugin } from 'workbox-expiration';
+import { offlineFallback } from 'workbox-recipes';
+import { setDefaultHandler, registerRoute } from 'workbox-routing';
+import { CacheFirst, NetworkFirst, NetworkOnly } from 'workbox-strategies';
+import {CacheableResponsePlugin} from 'workbox-cacheable-response';
+import { clientsClaim } from 'workbox-core';
+
+self.skipWaiting();
+clientsClaim();
+
+// Offline Page
+setDefaultHandler(
+    new NetworkOnly()
+);
+
+offlineFallback({
+    pageFallback: '/static/offline.html',
+    imageFallback: '/static/images/logo_OL-lg.png'
+});
+
+function matchFunction({ url }) {
+    const pages = ['/', '/account/login', '/account', '/account/books', '/account/loans', '/account/books/already-read/stats'];
+    return pages.includes(url.pathname);
+}
+
+// runtime caching as the user visits the page
+registerRoute(
+    matchFunction,
+    new NetworkFirst({
+        cacheName: 'html-cache',
+        plugins: [
+            new ExpirationPlugin({
+                // Keep at most 50 entries.
+                maxEntries: 50,
+                // Don't keep any entries for more than 30 days.
+                maxAgeSeconds: 30 * 24 * 60 * 60,
+                // Automatically cleanup if quota is exceeded.
+                purgeOnQuotaError: true,
+            }),
+            // only cache if it the request returns 0 or 200 status
+            new CacheableResponsePlugin({
+                statuses: [0, 200],
+            }),
+        ],
+    })
+);
+
+// covers png cache
+registerRoute(
+    new RegExp('https://covers.openlibrary.org/b/.+'),
```

Comment:
Covers can sometimes have a variety of paths here; let's do any .png

```suggestion
    new RegExp('https://covers.openlibrary.org/.+'),
```